### PR TITLE
Added documentation for DEALLOCATE statement

### PR DESCRIPTION
**Pull request by matriv**

### CODE REVIEW

**Comment by matriv** [comment]
@nslater @andreidan  Any suggestions for where we could add a reference to the new statement in https://github.com/crate/crate/blob/master/blackbox/docs/interfaces/postgres.rst are welcomed! :-)

File: blackbox/docs/sql/statements/deallocate.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    DEALLOCATE [PREPARE] { name | ALL }
+
+Description
+===========
+
+DEALLOCATE is used to deallocate a previously prepared SQL statement and free
+the reserved resources. It is not meant to be explicitly issued by a user but
+it's used by some clients (e.g. `libpq`_) over Postgres Wire Protocol as an
+alternative way of deallocating a prepared statement to protocol's ``Close (F)``
```
Comment (andreidan) [inline blackbox/docs/sql/statements/deallocate.rst]: "to the protocol's"

File: blackbox/docs/sql/statements/deallocate.rst [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+::
+
+    DEALLOCATE [PREPARE] { name | ALL }
+
+Description
+===========
+
+DEALLOCATE is used to deallocate a previously prepared SQL statement and free
+the reserved resources. It is not meant to be explicitly issued by a user but
+it's used by some clients (e.g. `libpq`_) over Postgres Wire Protocol as an
```
Comment (andreidan) [inline blackbox/docs/sql/statements/deallocate.rst]: @matriv maybe link "Postgres Wire Protocol" to  https://github.com/crate/crate/blob/master/blackbox/docs/interfaces/postgres.rst

Comment (matriv) [reply blackbox/docs/sql/statements/deallocate.rst]: I'm not sure... I was thinking to link from that file to here somehow... Waiting for @nslater  suggestion too.

Comment (nslater) [reply blackbox/docs/sql/statements/deallocate.rst]: yeah that looks fine to me

**Pull request merged**